### PR TITLE
Switch to using cluster-standup-teardown for WC creation

### PR DESCRIPTION
**Pull request by AverageMarcus**
### What this PR does

Switches to using [cluster-standup-teardown](https://github.com/giantswarm/cluster-standup-teardown) to handle the creation and deletion of workload clusters.

Towards: https://github.com/giantswarm/giantswarm/issues/30661

### Checklist

- [x] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,13 +2,15 @@ module github.com/giantswarm/cluster-test-suites
 
 go 1.21
 
+replace github.com/giantswarm/cluster-standup-teardown => github.com/giantswarm/cluster-standup-teardown v0.0.0-20240426074914-3870735834b4
```
Comment (AverageMarcus) [inline go.mod:5]: Currently required while testing this PR out. Once confirmed that this change works I can merge and release the initial version of `cluster-standup-teardown` and remove this replace directive.

**Comment by AverageMarcus** [comment]
The CAPZ is a known failure and caused by a problem with the dns-operator on our CAPZ MC.

File: go.mod (hunk 1 above)
Comment (AverageMarcus) [reply go.mod]: Done.

**Comment by AverageMarcus** [comment]
Module updated to use the released v1.0.0. Re-running tests to be sure.

/run cluster-test-suites

File: Dockerfile [hunk 2]
```diff
@@ -6,13 +6,10 @@ RUN go install github.com/onsi/ginkgo/v2/ginkgo@latest
 
 ADD go.mod go.sum ./
 
-RUN go mod tidy
+RUN go mod download
```
Comment (pipo02mix) [inline Dockerfile:9]: is this intended?

**Review by pipo02mix — APPROVED** [review]
did not review in deep but LGTMIIWFY

File: Dockerfile (hunk 2 above)
Comment (AverageMarcus) [reply Dockerfile:9]: Yes, it was wrong initially.

**Review by mcharriere — APPROVED** [review]
lgtm!

**Pull request merged**